ROX-18106: compliance profile protos stores

dashrews78 opened the pull request:
## Description

Just the protos and stores for profiles and rules.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI of generated UTs.

dashrews78 committed:
X-Smart-Branch-Parent: master

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -0,0 +1,43 @@
+    string              name            = 3 [(gogoproto.moretags) = 'search:"Compliance Profile Name,hidden,store" sql:"index=category:unique;name:rule_unique_indicator"'];
+    string              version         = 4 [(gogoproto.moretags) = 'search:"Compliance Operator Version,hidden,store" sql:"index=category:unique;name:rule_unique_indicator"'];
+    string              profile_version = 5 [(gogoproto.moretags) = 'search:"Compliance Profile Version,hidden,store"'];
+    string              product_type    = 6 [(gogoproto.moretags) = 'search:"Compliance Operator Profile Product,hidden,store"'];
```
1:1 mapping will not hold true in the case of composite profiles. I recommend an array.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -0,0 +1,43 @@
+        string rule_name       = 1 [(gogoproto.moretags) = 'search:"-" sql:"fk(ComplianceOperatorRuleV2:name),no-fk-constraint"'];
+    }
+    repeated Rule       rules       = 11;
+}
```
Let's capture the product as column (?)

dashrews78 replied:
I'm not tracking this question.  What do you mean by "product"?

md2119 replied:
`compliance.openshift.io/product: redhat_openshift_container_platform_4.1`

dashrews78 replied:
got it.  sure.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -0,0 +1,43 @@
+    string              name         = 1 [(gogoproto.moretags) = 'search:"Compliance Rule Name,hidden,store" sql:"pk"'];
+    string              version      = 2 [(gogoproto.moretags) = 'search:"Compliance Operator Version,hidden,store"'];
+    string              rule_version = 3 [(gogoproto.moretags) = 'search:"Compliance Rule Version,hidden,store"'];
+    string              product_type = 4 [(gogoproto.moretags) = 'search:"Compliance Operator Rule Product,hidden,store"'];
```
Are you planning on deriving `product_type` from profile?

dashrews78 replied:
I just took a closer look at the comments on the data model.  I'm going to take this out.  
When we get to doing tailored profiles we will have to re-evaluate if we need something there or not.

md2119 commented on proto/storage/compliance_operator_v2.proto:
```diff
@@ -0,0 +1,43 @@
+    string              version      = 2 [(gogoproto.moretags) = 'search:"Compliance Operator Version,hidden,store"'];
+    string              rule_version = 3 [(gogoproto.moretags) = 'search:"Compliance Rule Version,hidden,store"'];
+    string              product_type = 4 [(gogoproto.moretags) = 'search:"Compliance Operator Rule Product,hidden,store"'];
+    string              rule_type    = 5 [(gogoproto.moretags) = 'search:"Compliance Operator Rule Type,hidden,store"'];
```
What is a rule type?

dashrews78 replied:
Node, Platform, Workload, etc.  It is the `CheckType` field in the CR I'm looking at.

md2119 reviewed (commented):
I made some recommendations and left some clarifying questions.

dashrews78 committed:
X-Smart-Squash: Squashed 11 commits:

165cb511ff profile rules stores and protos
d0387b24f0 datastore
b1022cae93 some extra characters
92de045445 style
eeb586e99a terminate quote
be6c233d97 switch key
de25798169 style
0a057b01c1 adjust the repeated
4c81c7f075 fix test
2ae7a50a58 some rework
f9dc034c67 fix search field

dashrews78 committed:
rebase to regenerate
